aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp

```diff
@@ -281,8 +281,41 @@ int64_t hsum_sq(const uint8_t* A, int len) {
   int i = 0;
 
 #ifdef CPU_CAPABILITY_AVX2
-  __m256i sum_v_epu32 = _mm256_setzero_si256();
   // vectorized
+  __m256i sum_v_epu32 = _mm256_setzero_si256();
+  alignas(64) int32_t temp[8];
+
+  int overflow_threshold = 262144; // 2147483647(max of int32)/(256*256)*8 = 262144 
+  if( len > overflow_threshold){
+
+    int loop = len / overflow_threshold;
+
+    for(int j=1; j<=loop; j++){
+      for (; i < overflow_threshold * j; i += 16) {
```

Comment:
looks good conceptually, thanks for finding and fixing the issue!

If we start the outer loop from `j=0`, can we remove the loop in lines L319:L335 completely?  Might be a bit cleaner since it would remove duplication of the intinsic code.  Same for the int8_t version.

Reply:
let me know if you're interested in doing the cleanup above, or if you'd prefer to land as is (since this is an important numerics fix), and we can find someone else for the refactor.  Either way works, let me know your preference!

Reply:
Thanks @vkuzo for the review. I think the lines L319:L335 is focusing on the vectorized for the index from: `len/overflow_threshold*overflow_threshold` to `len`. If we start the outer loop from `j=0`, I still don't think we can clean these lines.